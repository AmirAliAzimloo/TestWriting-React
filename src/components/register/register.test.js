import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Register from "./index";

const getElement=(element)=>{
  const elements={
    "Email":screen.getByRole("textbox", { name: "Email"}),
    "Password": screen.getByLabelText("Password"),
    "Confirm Password": screen.getByLabelText("Confirm Password"),
  }
  if( elements[element])  return elements[element]
}
describe("Register Page", () => {
  test("Inputs should be initially empty", () => {

    // SingleEllement // like one input
    //getEllemnt => vaghti safhe az aval mikhaym yek elementi ro dashte bashe
    //queryEllement => vaghti bad az yek amalkardi mikhaym yek elementi ro dashte bashe
    //findEllement => vaghti mikhaym async/await ro handel konim

    // MultipleEllement // like options of dropdown
    // getAll...
    // queryyAll...
    // findAll...


    // Arrange
    render(
    <BrowserRouter>
    <Register/>
    </BrowserRouter>
    )

    // Act
    // Change event - click event or ...
    // Assert

    expect(getElement("Email").value).toBe("");
    expect(getElement("Password").value).toBe("");
    expect(getElement("Confirm Password").value).toBe("");
  });
});
