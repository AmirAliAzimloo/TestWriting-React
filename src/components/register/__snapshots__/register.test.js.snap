// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Page handle errors and navigate snappshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h1>
          Register To Login
        </h1>
        <div
          class="content"
        >
          <label
            for="email"
          >
            Email
          </label>
          <input
            id="email"
            name="email"
            type="email"
            value=""
          />
        </div>
        <div
          class="content"
        >
          <label
            for="password"
          >
            Password
          </label>
          <input
            id="password"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="content"
        >
          <label
            for="confirm-password"
          >
            Confirm Password
          </label>
          <input
            id="confirm-password"
            name="confirmPassword"
            type="password"
            value=""
          />
        </div>
        <div
          class="content"
        >
          <button
            class="btn"
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </div>
    <div>
      <div
        class="container"
      >
        <h1>
          Register To Login
        </h1>
        <div
          class="content"
        >
          <label
            for="email"
          >
            Email
          </label>
          <input
            id="email"
            name="email"
            type="email"
            value=""
          />
        </div>
        <div
          class="content"
        >
          <label
            for="password"
          >
            Password
          </label>
          <input
            id="password"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="content"
        >
          <label
            for="confirm-password"
          >
            Confirm Password
          </label>
          <input
            id="confirm-password"
            name="confirmPassword"
            type="password"
            value=""
          />
        </div>
        <div
          class="content"
        >
          <button
            class="btn"
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h1>
        Register To Login
      </h1>
      <div
        class="content"
      >
        <label
          for="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          type="email"
          value=""
        />
      </div>
      <div
        class="content"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          type="password"
          value=""
        />
      </div>
      <div
        class="content"
      >
        <label
          for="confirm-password"
        >
          Confirm Password
        </label>
        <input
          id="confirm-password"
          name="confirmPassword"
          type="password"
          value=""
        />
      </div>
      <div
        class="content"
      >
        <button
          class="btn"
          type="submit"
        >
          Submit
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
